podman_compose.py

```diff
@@ -1273,33 +1274,63 @@ async def container_to_args(compose, cnt, detached=True):
     return podman_args
 
 
-def rec_deps(services, service_name, start_point=None):
-    """
-    return all dependencies of service_name recursively
-    """
-    if not start_point:
-        start_point = service_name
-    deps = services[service_name]["_deps"]
-    for dep_name in deps.copy():
-        # avoid A depens on A
-        if dep_name == service_name:
-            continue
-        dep_srv = services.get(dep_name, None)
-        if not dep_srv:
-            continue
-        # NOTE: avoid creating loops, A->B->A
-        if start_point and start_point in dep_srv["_deps"]:
-            continue
-        new_deps = rec_deps(services, dep_name, start_point)
-        deps.update(new_deps)
-    return deps
+class ServiceDependencyCondition(Enum):
+    NONE = ""
+    HEALTHY = "service_healthy"    
+    STARTED = "service_started"
+    COMPLETED = "service_completed_successfully"
+
+
+class ServiceDependency:
+    def __init__(self, name, condition=None):
+        self._name = name
+        self._condition = ServiceDependencyCondition(condition or "")
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def condition(self):
+        return self._condition
+
+    def __hash__(self):
+        # Compute hash based on the frozenset of items to ensure order does not matter
+        return hash(('name', self._name)+('condition', self._condition))
+
+    def __eq__(self, other):
+        # Compare equality based on dictionary content
+        if isinstance(other, ServiceDependency):
+            return self._name == other.name and self._condition == other.condition
+        return False
 
 
 def flat_deps(services, with_extends=False):
     """
     create dependencies "_deps" or update it recursively for all services
     """
+    def rec_deps(services, service_name, start_point=None):
```

Comment:
Moving this function made reviewing way harder.

Reply:
Noted, but... do you want me to move it out? Because it makes full sense to have it there...

Reply:
I would say it makes things more confusing. When functions are moved to other functions it's often because they need to refer to variables of parent function. The dependencies are less clear.

Reply:
Ok: I will move it out, because it's not the point to have a discussion about this :-) You are right when you say that "it's often because they need to refer to variables of parent function" (although this is not the case here), but inner functions are also used when you want to encapsulate functionality on its own function but still not make it public to be called from anywhere else.